services-core/src/solution_submission/retry.rs

```diff
@@ -149,24 +127,37 @@ impl<'a> RetryWithGasPriceIncrease<'a> {
         RetryResult::Cancelled(result.map(|_| ()))
     }
 
-    // Yields the current gas price immediately and then every refresh interval.
-    fn gas_price_stream(
-        &self,
-        target_confirm_time: Instant,
-    ) -> impl FusedStream<Item = Result<f64>> + '_ {
+    // Yields the current gas price immediately and then every refresh interval. Skips errors.
+    fn gas_price_stream(&self, target_confirm_time: Instant) -> impl FusedStream<Item = f64> + '_ {
         stream::unfold(true, move |first_call| async move {
             if !first_call {
                 self.async_sleep.sleep(GAS_PRICE_REFRESH_INTERVAL).await;
             }
             return Some((self.gas_price(target_confirm_time).await, false));
         })
+        .filter_map(|gas_price_result| async move {
+            match gas_price_result {
+                Ok(gas_price) => {
+                    log::debug!("estimated gas price {}", gas_price);
+                    Some(gas_price)
+                }
+                Err(err) => {
+                    log::error!("gas price estimation failed: {:?}", err);
+                    None
+                }
+            }
+        })
     }
 
     async fn retry_(&self, args: Args, cancel_after: impl Future) -> RetryResult {
         log::debug!("starting retry with gas price cap {}", args.gas_price_cap);
 
-        let gas_price_stream = self.gas_price_stream(args.target_confirm_time);
+        let gas_price_stream = gas_price_increase::enforce_minimum_increase(
```

Comment:
nit: Syntactically, I would find it awesome if `enforce_minimum_increase` could be an extension on the Stream to allow this line to look like:

```rust
let gas_price_stream = self
  .gas_price_stream(args.target_confirm_time)
  .enforce_minimum_increase(args.gas_price_cap);
```

Reply:
Not a fan because then we have to introduce a new trait and implement this trait for Stream<Item=f64>. I feel this makes the code overall harder to understand just for slightly different syntax here.

Reply:
I also think its not worth it in this case just for some syntactic sugar. If the `gas_price_stream` returned a specific `GasPriceStream: Stream` type, then I 100% agree it would make sense to implement this as a method.